### PR TITLE
Rotate Kick Development

### PR DESCRIPTION
**Pull request by Squid5678**
## Description
Goal: When the robot is in possession of the ball, we want the robot to be able to rotate in-place (effectively, pivot) while holding the ball, and then score. 

## Associated / Resolved Issue
Link to ClickUp Card: https://app.clickup.com/t/86b1vtxnx

NOTE: As of right now, there is a current issue with path-target. When the robot initially plans towards the ball, the ending velocity of the ball (or the ending position of the robot when path planning to the ball) results in the robot losing contact to the ball after planning to the ball. This issue occurs "at random" and is attributed to a separate issue with the planning to the ball itself.

### CODE REVIEW

File: docs/source/tutorial.rst [hunk 1]
```diff
@@ -268,109 +268,7 @@ the runner (and every other robot on our team) move much more quickly.
 Take a screen recording of this whole process and send it to your software lead
 via Slack. Feel free to play around with any other params you see!
 
-5. Action Clients and building a position
```
Comment (jacksherling) [inline docs/source/tutorial.rst]: commit issue?

File: soccer/src/soccer/planning/planner/rotate_path_planner.cpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            // SPDLOG_INFO("reset");
+            plan_angles(&path, start_instant, AngleFns::face_point(pivot_target), request.constraints.rot);
+            // plan_angles(&path, start_instant, func, request.constraints.rot);
+            path.stamp(RJ::now());
+            cached_path_ = path;
+        }
+    } else {
+        // SPDLOG_INFO("reset");
+        cached_path_.reset();
+        plan_angles(&path, start_instant, AngleFns::face_point(pivot_target), request.constraints.rot);
```
Comment (jacksherling) [inline soccer/src/soccer/planning/planner/rotate_path_planner.cpp]: A lot of repeated code in this case from above. Is there a way we can get clever with the if lgoic? no worries if not

File: soccer/src/soccer/planning/planner/rotate_path_planner.cpp [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    // SPDLOG_INFO("cached_target_agne: {}", cached_target_angle_.value_or(0));
+    // SPDLOG_INFO("target angle: {}", target_angle);
+
+    cached_angle_change_ = angle_change;
+    // SPDLOG_INFO("cached angle change: {}", *cached_angle_change_);
+
+
+    Trajectory path{};
+
+    if (abs(*cached_target_angle_ - target_angle) < degrees_to_radians(1)) {
```
Comment (jacksherling) [inline soccer/src/soccer/planning/planner/rotate_path_planner.cpp]: Magic number? Is this supposed to be kIsDoneAngleChangeThresh?

File: soccer/src/soccer/planning/primitives/angle_planning.cpp [hunk 4]
```diff
@@ -6,11 +6,35 @@ namespace planning {
...
+        }
+
+        if (delta_angle < -M_PI) {
+            delta_angle += 2 * M_PI;
+        }
+
+        double time = Trapezoidal::get_time(abs(delta_angle), abs(delta_angle), constraints.max_speed, constraints.max_accel, 0, 0);
+        
+
+        SPDLOG_INFO("delta angle: {}", delta_angle);
```
Comment (jacksherling) [inline soccer/src/soccer/planning/primitives/angle_planning.cpp]: probably want to remove this

File: docs/source/tutorial.rst (hunk 1 above)
Comment (Squid5678) [reply docs/source/tutorial.rst]: Yeah, something happened, it should be there with this new commit.

File: soccer/src/soccer/planning/planner/rotate_path_planner.cpp (hunk 3 above)
Comment (Squid5678) [reply soccer/src/soccer/planning/planner/rotate_path_planner.cpp]: fixed

File: soccer/src/soccer/planning/primitives/angle_planning.cpp (hunk 4 above)
Comment (Squid5678) [reply soccer/src/soccer/planning/primitives/angle_planning.cpp]: done

**Review by sanatd33 — APPROVED** [review]

**Review by jacksherling — APPROVED** [review]

**Pull request merged**